flag_groups_test.go

```diff
@@ -183,13 +199,39 @@ func TestValidateFlagGroups(t *testing.T) {
 				sub.MarkFlagsMutuallyExclusive(strings.Split(flagGroup, " ")...)
 			}
 			c.SetArgs(tc.args)
-			err := c.Execute()
+			executedCmd, err := c.ExecuteC()
 			switch {
 			case err == nil && len(tc.expectErr) > 0:
 				t.Errorf("Expected error %q but got nil", tc.expectErr)
 			case err != nil && err.Error() != tc.expectErr:
 				t.Errorf("Expected error %q but got %q", tc.expectErr, err)
 			}
+
+			if len(tc.expectErr) > 0 {
+				var flagGroupErr *FlagGroupError
+				if !errors.As(err, &flagGroupErr) {
```

Comment:
Please read this

https://github.com/spf13/cobra/pull/2266#discussion_r2074792615